apps/tangle-dapp/components/LiquidStaking/stakeAndUnstake/UnstakePeriodDetailItem.tsx

```diff
@@ -1,18 +1,61 @@
-import { FC } from 'react';
+import { SkeletonLoader } from '@webb-tools/webb-ui-components';
+import assert from 'assert';
+import { FC, useMemo } from 'react';
 
+import { ParachainCurrency } from '../../../constants/liquidStaking';
+import useTokenUnlockDurations from '../../../data/liquidStaking/useTokenUnlockDurations';
+import stringifyTimeUnit from '../../../utils/liquidStaking/stringifyTimeUnit';
 import DetailItem from './DetailItem';
 
-const UnstakePeriodDetailItem: FC = () => {
-  // TODO: Load this info from the chain. Currently using dummy data.
+export type UnstakePeriodDetailItemProps = {
+  currency: ParachainCurrency;
+};
+
+const UnstakePeriodDetailItem: FC<UnstakePeriodDetailItemProps> = ({
+  currency,
+}) => {
+  const unlockDurations = useTokenUnlockDurations();
+
+  const unlockPeriod = useMemo<'Unknown' | [string, string] | null>(() => {
+    if (unlockDurations === null) {
+      return null;
+    }
+
+    const unlockDuration = unlockDurations.find(
+      (entry) => entry.currency === currency && entry.isNative,
+    );
+
+    // Unlock duration is not set onchain.
+    if (unlockDuration === undefined) {
+      return 'Unknown' as const;
+    }
+
+    const parts = stringifyTimeUnit(unlockDuration.timeUnit).split(' ');
+
+    assert(parts.length === 2, 'Unexpected time unit format');
```

Comment:
Consider returning a tuple instead of using `assert` directly on the component, as it suggests the potential for failure and could crash the component.